fiftyone/utils/open_clip.py

```diff
@@ -0,0 +1,122 @@
+"""
+CLIP model wrapper for the FiftyOne Model Zoo.
+
+| Copyright 2017-2023, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import logging
+import os
+from packaging.version import Version
+import warnings
+
+import eta.core.web as etaw
+
+import fiftyone as fo
+import fiftyone.core.models as fom
+import fiftyone.core.utils as fou
+import fiftyone.utils.torch as fout
+import fiftyone.zoo.models as fozm
+
+fou.ensure_torch()
+import torch
+
+import open_clip
+
+logger = logging.getLogger(__name__)
+
+
+class TorchOpenClipModelConfig(fout.TorchImageModelConfig, fozm.HasZooModel):
+    """Configuration for running a :class:`TorchCLIPModel`.
+
+    See :class:`fiftyone.utils.torch.TorchImageModelConfig` for additional
+    arguments.
+
+    Args:
+        context_length: the model's context length
+        text_prompt: the text prompt to use, e.g., ``"A photo of"``
+        classes (None): a list of custom classes for zero-shot prediction
+        model_name_clip (None): the model to use
+        pretrained (None): the pretrained version to use
+    """
+
+    def __init__(self, d):
+        d = self.init(d)
+        super().__init__(d)
+
+        self.context_length = self.parse_int(d, "context_length")
+        self.text_prompt = self.parse_string(d, "text_prompt")
+
+        self.model_name_clip = self.parse_string(d, "model_name_clip")
```

Comment:
are `model_name` and `pretrained` required? If the OpenCLIP package defaults to something, you could update to this:

```py
self.model_name_clip = self.parse_string(d, "model_name_clip", default=None)
self.pretrained = self.parse_string(d, "pretrained", default=None)
```

so that users could just do this:

```py
model = foz.load_zoo_model("open-clip-torch")
```

without needing to know a particular pretrained model to load.

Reply:
Hi @brimoor,

Thanks a lot for your feedback! I'm working on the adjustments right now. Just to clarify, the 'pretrained' parameter is optional, but specifying the model name is necessary. To streamline the process, I'm setting the default model to 'ViT-B-32'. This should make it more user-friendly while ensuring essential parameters are in place.